oasislmf/exposures/csv_trans.py

```diff
@@ -6,65 +6,132 @@
 
 import json
 import logging
+import multiprocessing
 import os
 
 import pandas as pd
 
 from lxml import etree
 
+from oasislmf.utils.concurrency import (
+    multiprocess,
+    multithread,
+    Task,
+)
 
 class Translator(object):
-    def __init__(self, input_path, output_path, xsd_path, xslt_path, append_row_nums=False, chunk_size=5000, logger=None):
+    """
+    Transforms exposures/locations in CSV format
+    by converting a source file to XML and applying an XSLT transform
+    to apply rules for selecting, merging or updating columns
+
+    An optional step is to passing an XSD file for output validation
+
+    :param input_path: Source exposures file path, which should be in CSV comma delimited format 
+    :type input_path: str
+
+    :param output_path: File to write transform results
+    :type output_path: str
+
+    :param xslt_path: Source exposures Transformation rules file
+    :type xslt_path: str
+
+    :param xsd_path: Source exposures output validation file
+    :type xsd_path: str
+
+    :param append_row_nums: Append line numbers to first column of output called `ROW_ID` [1 .. n] when n is the number of rows processed. 
+    :type append_row_nums: boolean
+
+    :param chunk_size: Number of rows to process per multiprocess Task
+    :type chunk_size: int
+    """
+
+    def __init__(self, input_path, output_path, xslt_path, xsd_path=None, append_row_nums=False, chunk_size=5000, logger=None):
         self.logger = logger or logging.getLogger()
-        self.xsd = etree.parse(xsd_path)    # validation file
-        self.xslt = etree.parse(xslt_path)  # transform file
-        self.fpath_input = input_path       # file_in.csv
-        self.fpath_output = output_path     # file_out.csv
+        self.fpath_input = input_path
+        self.fpath_output = output_path
+        self.xslt = etree.parse(xslt_path)
+        self.xsd = (etree.parse(xsd_path) if xsd_path else None)
 
-        self.row_nums = append_row_nums  # Add 'ROW_ID' field to output
-        self.row_limit = chunk_size      # MAX number of CSV input rows
-        self.row_header_in = None        # CSV col header
-        self.row_header_out = None       # CSV col header Post Transform
+        self.row_nums = append_row_nums
+        self.row_limit = chunk_size
+        self.row_header_in = None
+        self.row_header_out = None
 
     def __call__(self):
         csv_reader = pd.read_csv(self.fpath_input, iterator=True, dtype=object, encoding='utf-8')
-        for data, first_row_number, last_row_number in self.next_file_slice(csv_reader):
-            self.logger.debug('--- lines[%d .. %d] ---', first_row_number, last_row_number)
 
-            # Convert CSV -> XML
-            xml_input_slice = self.csv_to_xml(
-                self.row_header_in,
-                data
+        task_list = []
```

Comment:
Just an observation but as the chunk processing doesn't seem to involve any filesystem access you could use `multiprocess` which uses processes rather than `multithread` which uses plain threads. It'll probably be quicker.